remoteappmanager/docker/container_manager.py

```diff
@@ -94,44 +90,29 @@ def stop_and_remove_container(self, container_id):
             self._stop_pending.remove(container_id)
 
     @gen.coroutine
-    def containers_for_image(self, image_id_or_name, user_name=None):
-        """Returns the currently running containers for a given image.
-
-        If `user_name` is given, only returns containers started by the
-        given user name.
-
-        It is a coroutine because we might want to run an inquire to the docker
-        service if not present.
+    def container_from_mapping_id(self, user_name, mapping_id):
+        """Returns the currently running container for a given user and
+        mapping_id.
 
         Parameters
         ----------
-        image_id_or_name: str
-            The image id or name
-
-        Optional parameters
-        -------------------
-        user_name : str
-            Name of the user who started the container
+        user_name: str
+            The username
+        mapping_id : str
+            The unique id to identify the container
 
         Return
         ------
-        A list of container objects, or an empty list if not present.
+        A container objects, or None if not present.
         """
-        if user_name:
-            user_labels = _get_container_labels(user_name)
-            if user_labels:
-                filters = {'label': '{0}={1}'.format(*user_labels.popitem())}
-        else:
-            filters = {}
-
-        filters['ancestor'] = image_id_or_name
+        labels = _get_container_labels(user_name, mapping_id)
+        filters = {
+            'label': ['{0}={1}'.format(k, v) for k, v in labels.items()]
+        }
 
         containers = yield self.docker_client.containers(filters=filters)
-        return [Container.from_docker_dict(container)
-                for container in containers
-                # Require further filtering as ancestor include grandparents
-                if (container.get('Image') == image_id_or_name or
-                    container.get('ImageID') == image_id_or_name)]
+        if len(containers) != 0:
+            return Container.from_docker_dict(containers[0])
```

Comment:
I don't think we are considering the case at all. The user is not allowed to start a container with the same configuration multiple times.